### PR TITLE
Improve the read me and fix the import issues

### PR DESCRIPTION
**Pull request by Moasib-Arif**
Changes:
- Added the Instructions how to install the repo from Github to local Machine (ReadMe File)
- Issues with imports (http and Config )fixed / All the import work

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[[source]]
```
Comment (mikeAdamss) [inline Pipfile]: we wont need pipfile or pipfile.lock, there are two very common but overlapping virtual environment managers for python:

- pipenv: which you know, allowed us to manage virtual environments easily.
- poetry: which is a python package manager _for creating installable python packages_ that has a virtual environment manager build in.

since we're using poetry here we don't need pipenv, they overlap.

Comment (mikeAdamss) [reply Pipfile]: both tools borrow from each other heavily, poetry uses pyproject.toml instead of Pipfile and poetry.lock instead of pipfile.lock but they perform similar tasks in regards to managing dependencies.

so:

- `poetry add <package>` rather than `pipenv install <package>`
- and `poetry shell` rather than `pipenv shell`
- and `poetry run <something>` rather than `pipenv run <something>`

a brief explanation of why the change:

 - Where we're creating an installable python package then you'd use poetry for dependancy management as poetry will be in use anyway.
- Where we _not_ creating an insallable python package (like where we were building an api) pipenv is preferable, otherwise you end up with odd details (like python package release number) in a pyproject.toml which implies its a python package even though its not.

TLDR:
installable python package - poetry
just managing a virtual environment - pipenv

right tool for the right job etc.

File: Pipfile.lock
Comment (mikeAdamss) [inline Pipfile.lock:1]: can go as well.

File: dpytools/__init__.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-from http.http import HttpClient
+from http_client.http import HttpClient
```
Comment (mikeAdamss) [inline dpytools/__init__.py]: try and avoid the word client singular please, this will end up as something like

```
/base_client.py. HttpClient
/datasets-api-client.py.  DatasetApiClient (inherits from HttpClient)
/some-other-api.client.py SomeOtherApiClient (also inherits from HttpClient)
```

so it'll be a directory of http client(s) so a developer can import a batteries included mechanism for interacting with whichever api they're interacting with.

File: README.md [hunk 3]
```diff
@@ -4,7 +4,23 @@ Simple reusable python resources for digital publishing.
 
 ## Installation
 
-**TODO**
+Follow these steps to install the package on your local machine:
```
Comment (mikeAdamss) [inline README.md:7]: This is correct information but there's a much easier way to accomplish this. Have a look at this: https://www.geeksforgeeks.org/how-to-install-a-python-package-from-a-github-repository/

You should be able to get this down to something _like_ `pip install git+http://<something, cant remember>`.

Literally a one liner to install the python package from a github repo in the same way you would from pypi.

Take some time on understanding that one, its a phenomanally useful trick for a python dev to know.

From memory you should be able to specify a branch in there as well, so should be able to install from your dev branch to try it out - have a play 😄

**Review by mikeAdamss — COMMENTED** [review]
A few tweaks to make around dependencies and installation but nothing major, give me a shout if I've explained badly.

RE test failure, its just failing as its looking to produce code coverage but no tests exist to run so it cant find any results.

for now just add a `/tests/test_nothing.py` file containing:

```

# TODO - remove as soon as we add our first test.
def test_nothing():
    """
    Empty test to stop failures for 0 coverage
    """
    ...
```

And that'll move us to a green tick.
